test/test_files/cast/cast_to_nested_types.test

```diff
@@ -372,3 +372,27 @@ Conversion exception: Map does not allow null as key.
 -STATEMENT RETURN cast(cast([220], "decimal(21,4)[]"), "int[1]")
 ---- 1
 [220]
+
+-CASE CastToUnion
+-STATEMENT CREATE NODE TABLE Person(id INT64, description UNION(height INT64, name STRING), PRIMARY KEY(id))
```

Comment:
Let's test the above cases:
1. Unflat case (where your params has selSize > 1)
```
match (p:person) return cast(p.id as union(a int32, b string));
```
where you have multiple people in your table.
2. Add a test case where your input type can be cast to multiple fields in the union.
```
return cast(127 as union(a int64, b int32))
return cast(3.2 as union(a float, b double))
```
3. We should error on casting nested types to union(since they are not supported right now)

Reply:
Done with https://github.com/kuzudb/kuzu/pull/5607/commits/603618afc4253d4e21514b227ca4ec7b064296e1